ideascube/serveradmin/catalog.py

```diff
@@ -351,18 +351,31 @@ def _get_package(self, id, source):
         except KeyError:
             raise InvalidPackageType(type)
 
-    def _get_packages(self, id_patterns, source, fail_if_no_match=True):
+    def _get_packages(
+        self, id_patterns, source,
+        fail_if_no_match=True, fail_if_invalid_package=True):
```

Comment:
Having the first `fail_if_no_match` boolean already was something I found pretty bad... when I introduced it.

I agree that this code needs refactoring, but maybe let's hold off on it a bit, because lots of things will need to change when we start using the catalog in the web app anyway. (multiple simultaneous accesses anyone?)